crates/api/src/peer_store.rs

```diff
@@ -56,6 +56,16 @@ pub trait PeerStore: 'static + Send + Sync + std::fmt::Debug {
         &self,
         peer_url: Url,
     ) -> BoxFut<'_, K2Result<Vec<Arc<AgentInfoSigned>>>>;
+
+    /// Register a listener that will be called whenever a peer is added or updated in the store.
+    fn register_peer_update_listener(
+        &self,
+        _listener: Arc<
+            dyn (Fn(Arc<AgentInfoSigned>) -> BoxFut<'static, ()>) + Send + Sync,
```

Comment:
Inserting agents into the peer store is done with a list of agent infos and happens at the frequency of querying the bootstrap and through gossip and broadcast. Removing agents is only done when blocking them and won't happen that frequently. Is it better to make this a list of agents too to not have to loop over the list of agents to call the listener, which requires locking the RwLock each time?

Reply:
Ah yes, that doesn't make a difference.